### ISSUE TITLE
group by fact

### ISSUE THREAD

**Issue by anarcat** [sodabrew/puppet-dashboard#321]
one thing that puppet-board does and puppet-dashboard doesn't is to allow grouping results by fact.

for example, this screenshot shows a report listing the puppetversion of all nodes in one view:

![screenshot of puppetboard showing all nodes's puppetversion fact](https://camo.githubusercontent.com/d5ef188308f60468fbbf35bcb77c7f15a20e1a82/68747470733a2f2f7261772e6769746875622e636f6d2f6e656461702f707570706574626f6172642f6d61737465722f73637265656e73686f74732f666163742e706e67)

is this something that works in puppet-dashboard or that is planned?

thanks

**Comment by sodabrew** [comment]
Awesome feature request! Pull requests welcomed!

I have more time available to help code review such a feature than I do to write it myself at the moment.

**Comment by anarcat** [comment]
any advice on where one would start coding this? or UX recommndations should this be a new tab? i initially tried to make the "inventory search" do the right thing but it has to take a value for the fact and it displays only a static set of columns...

**Comment by sodabrew** [comment]
Is this for the Inventory view of nodes, or for displaying Variables across multiple nodes/classes/groups?

**Comment by anarcat** [comment]
i don't quite care where it's done, really. :) the idea is to display the value of a single (or even better, multiple) fact(s) across multiple nodes.

**Comment by arnaudmm** [comment]
Let me elucidate something... To do that, puppetdashboard has to talk to puppetdb api hasn't it ? 

Because puppetdashboard can't storeconfig, puppetdb is here for that.

Correct me if I'm wrong but I don't think puppetdashboard is (yet) puppetdb aware..

If you consider to do so, reports can also be stored (an retrieved) in(from) puppetdb. I don't know if it's a good idea. I'm currently using puppetdashboard as an ENC and puppetboard for digging into puppetdb datas (and import/export resources into it)

"Puppet enterprise dashboard" actually is a mix of your puppet dasbhoard and puppetboard. So I think a call has to be done here:

- use puppetdashboard for ENC fonctionnality only AND use puppetboard for catalog/fact/reports explorations.

- or try to "merge" projects to do the same than puppet enterprise dashboard

What do you think about this ? @sodabrew

**Comment by anarcat** [comment]
hmm... i believe puppet-dashboard already talks to puppetdb, at least i had to install it as a dependency when i setup puppet-dashboard.

no?

**Comment by sodabrew** [comment]
Negative, puppet dashboard doesn't talk to puppetdb at this time. Pull requests welcome!!

**Comment by arnaudmm** [comment]
Yeah that is what I was thinking. Would be great to have a project doing
both : getting informations from puppetdb and being an ENC :-)

It's a big change and it comes with design consideration

2015-05-05 22:38 GMT+02:00 Aaron Stone <notifications@github.com>:

**Comment by anarcat** [comment]
that's really weird: for some reason i configured puppetdb first here:

http://isuma-media-players.readthedocs.org/en/latest/configmanagement.html#configuring-a-dashboard

maybe i was confused with puppetboard? anyways, it seems it can run alongside puppetdb fine right now :)
